expyfun/_eyelink_controller.py

```diff
@@ -528,30 +528,105 @@ def wait_for_fix(self, fix_pos, fix_time=0., tol=100., max_wait=np.inf,
 
         return fix_success
 
+    def maintain_fix(self, fix_pos, check_duration, tol=100., period=.250,
+                     check_interval=0.001, units='norm', stop_early=False):
+        """Check to see if subject is fixating in a region.
+
+        This checks to make sure that the subjects gaze falls within a region
+        at least once in any given period. In other words it checks to make
+        sure that the gaze falls in the target region at least every
+        `period` seconds.
+
+        Parameters
+        ----------
+        fix_pos : tuple (length 2)
+            The screen position (in specified units) required.
+        check_duration : float
+            Total amount of time to check.
+        tol : float
+            The tolerance (in pixels) to consider the target hit.
```

Comment:
We usually allow units for these sorts of things. Maybe this should be in the same units as `fix_pos`. And is it a radius (circular area) or w/h (square)? (I should be able to tell from the var name ideally, and if not there, then the description.)

Reply:
I was using ``wait_for_fix()`` as a template which has allows any units for fix_pos but _only_ does tolerance in pixels.

Reply:
Do we want to change both ``wait_for_fix()`` and ``maintain_fix()`` to allow for any units?

Reply:
Hmm, that's a good question. Does it seem worth it to you?

Reply:
Well, the issue is that the tolerances in pixels make sense anywhere on the screen but if you have fixation dots off to the side and are trying to use degrees as tolerances, things get a little funky. So if we allowed for any units I'd say we'd end up converting whatever tolerances in degrees or norm to pixels at the center and then use those.

I think it's worth it as long as we're transparent about the units not being true on the sides.

Reply:
Good point. We can keep pixels for now, then, and add units later if we need them